core/src/main/kotlin/com/malinskiy/marathon/execution/device/DeviceActor.kt

```diff
@@ -15,22 +16,31 @@ import com.malinskiy.marathon.execution.withRetry
 import com.malinskiy.marathon.log.MarathonLogging
 import com.malinskiy.marathon.test.TestBatch
 import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.Job
 import kotlinx.coroutines.async
+import kotlinx.coroutines.cancelAndJoin
 import kotlinx.coroutines.channels.SendChannel
+import kotlinx.coroutines.delay
 import kotlinx.coroutines.isActive
 import kotlinx.coroutines.launch
 import kotlin.coroutines.CoroutineContext
 import kotlin.properties.Delegates
 
-class DeviceActor(private val devicePoolId: DevicePoolId,
-                  private val pool: SendChannel<DevicePoolMessage>,
-                  private val configuration: Configuration,
-                  val device: Device,
-                  private val progressReporter: ProgressReporter,
-                  parent: Job,
-                  context: CoroutineContext) :
-        Actor<DeviceEvent>(parent = parent, context = context) {
+class DeviceActor(
+    private val devicePoolId: DevicePoolId,
+    private val pool: SendChannel<DevicePoolMessage>,
+    private val configuration: Configuration,
+    val device: Device,
+    private val progressReporter: ProgressReporter,
+    parent: Job,
+    context: CoroutineContext
+) :
+    Actor<DeviceEvent>(parent = parent, context = context) {
+
+    companion object {
+        private const val AWAIT_BATCH_RESULTS_TIMEOUT_MS = 500L
```

Comment:
This looks like a vendor-specific value. I think it should be defined in configuration, or at least as an overridable property of device.

Reply:
No, it is not related in any way to the vendor - if we can't get deferred results in 500ms (when the device is terminated) let's terminate waiting and return batch to the queue. We can discuss the best value for both platforms.